### DISCUSSION TITLE
GitHub enterprise license usage for an organization

### DISCUSSION THREAD

**Discussion by nmattamGitHub** [community/community#23678] [Enterprise]
<p>Hi there, I need help listing the licenses consumed for an organization in our enterprise. Our enterprise has multiple organizations, and when I see the licenses consumed under Settings -&gt; Enterprise Licenses, I see GitHub lists all the licenses for all the organizations. Is there a better way to separate or list the licenses consumed for each organization individually under the enterprise? Please suggest.</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/nmattamgithub">@nmattamGitHub</a>, Enterprise accounts license unique users within the enterprise rather than users within an organization. If it did the latter and a user existed in more than one organization the number of licenses and costs would increase greatly.</p>
<p>So getting the list of members of your enterprise is correct.<br>
You can query users and all organizations they are a member of, but for reasons noted above you cannot attribute a user license to a specific organization (unless a user is a member of a single organization only)</p>

**Comment by nmattamGitHub** [comment]
<p>Thanks <a class="mention" href="/byrneh">@byrneh</a> for the details, it makes sense. Just in case a user is a member of a single organization what is the best way we can query the licenses consumed for the organization. Is it just GitHub interface that provides the details or are there any better approaches to it?</p>
<p>One reason for viewing the licenses at Org level is, we may want to restrict license usage, for example If Org A is allotted 20 licenses and Org B 50 licenses, Org A should only be utilizing its available 20 licenses and not from Org B if any additional licenses are required by Org A. Hence the ask. Sorry for being ignorant.</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/nmattamgithub">@nmattamGitHub</a>, you have nt actually mentioned whether you are referring to enterprise server (on premise) or Enterprise Cloud (sitting on <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>).<br>
You can query the members of an organization using GitHub’s rest REST API v3 or GraphQL API v4. AS already stated a unique under and enterprise account will be one license, so if a user is a member of more than one organization  only one enterprise user licenses is consumed.</p>
<p>Here is an example of a rest api<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/orgs#list-organization-members" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/orgs#list-organization-members" target="_blank" rel="noopener nofollow ugc">Organizations - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<br>
If you run into problems using API and querying the data you need, it would be best to cerate a new question in the relevant topic.
<p>You can mark this as solved if your question is now answered.</p>
